server/utils/cloudFunctions/functions/test/algoliaTests/algoliaIntegrationTests.test.js

```diff
@@ -0,0 +1,242 @@
+ *       error, the test may still pass. Checking for a `null` field
+ *       and a string equality match is alternatively much easier. 
+ *     - For the above reason, `getObjects` is used with singleton lists
+ *       to verify that a record has been deleted instead of `getObject`. 
+ * - `setTimeout` is used to allow firebase changes to propogate to algolia. 
+ *   - `setTimeout` requires that `done()` be called at the end of the 
+ *     inner function body. It is not equivalent to `return`. 
+ *   - Due to the use of `done()` at the end of `setTimeout`, promises
+ *     are not returned as they are in `countAdvertisersFunction` test suite.
+ *   - This is because it will make the test `overspecified`
+ *   - Read more here: `https://medium.com/@durja/mocha-tests-with-async-await-2aead4afeca1`
+ *   - This does not impact testing at all, but is a notable style difference.
+ * - `TIMEOUT_10S` was chosen because this is the maximum amount of latency 
+ *   for firebase to execute a function. 
+ * - `TIMEOUT_15S` is used to extend the default 2000ms test limit set by 
+ *   mocha.
+ * - The overall test timeout limit must be longer than the `setTimeout`
+ *   method to allow time for the algolia api call.    
+ * Tests are structured with promise nesting. 
+ * - There is a debate about nesting vs chaining promises. In this case, 
+ *     nesting is advantageous because it easily keeps everything in scope
+ *     without much fuss. Thus, nesting is fine to use. 
+ * - Also this is how the official docs do it. 
+ * - `Promise.allSettled` is used to avoid nesting hell where possible. 
+ *   Sometimes not possible if their is some sequential order that
+ *   must be obeyed (e.g., create and then delete.)  
+ * - Looked into using `await/async` but ran into more flakiness for some
+ *   reason. 
+ * Errors are handled directly by firebase, so their is no need to validate
+ * how errors are handled/resolved.   
+ * Author: Robert Marcus
+ * Date: July 16, 2020
+ */
+
+
+// Import the testing environment configuration.
+// `TIMEOUT_10S` is the maximum amount of latency for firebase 
+// to execute a function. 
+// `TIMEOUT_15S` is used to extend the default 2000ms test limit set by 
+// mocha. This gives a 5000ms latency period for the algolia api call
+// and corresponding tests to resolve and execute.
+const { chai,
+        TIMEOUT_10S,
+        TIMEOUT_15S,
+        DEV_ADS_COLLECTION } = require('../testConfig');
+
+const { deleteCollection } = require('../deleteCollection');
+
+// Import firebase environment configuration.
+const { DB } = require('../../firebaseConfig');
+
+// Import algolia dev environment configuration.
+const { DEV_ADS_INDEX } = 
+    require('../../algoliaCloudFunctions/algoliaConfig');
+
+describe("Algolia integrations tests", () => {
+  // After every 'describe' block, reset the test environments. 
+  // Since these are live dev DBs, this process can be very flaky.
```

Comment:
If deletion is flaky, tests should never be dependent on a previous deletion being successful.

Make each test case unique, delete on best-effort just to conserve space, and in practice you would set up an alert to yell at you when the DB got too big so you could go wipe it manually.

As I said earlier you should be able to achieve this by checking the system clock.  If that doesn't work you could also use RNG to assign a random value to insert at the start of every name.